lagom/composition.py

```diff
@@ -0,0 +1,56 @@
+from itertools import chain
+from typing import TypeVar, Type, Optional, Set, Tuple, Collection, Sequence
+
+from .container import Container
+from .interfaces import DefinitionsSource, SpecialDepDefinition
+
+X = TypeVar("X")
+
+
+def compose(*args: DefinitionsSource) -> Container:
+    combined_sources = _SourceList(*args)
+    _assert_unique_dependencies(args)
```

Comment:
Asserting uniqueness only when creating the composed container is probably unsafe, as the containers can be updated later on, no?
At least, I feel like there should be a test for that specific scenario.